src/cols.py

```diff
@@ -0,0 +1,77 @@
+import re
+from  src.Num import Num
+from  src.sym import Sym
+
+'''
+Class `Cols` Summarizes the first row of the dataset
+'''
+
+class Cols:
+    '''
+    Constuctor which initializes the attributes
+
+    Attributes:
+    @names: column names
+    @all: storing Num or Sym objects
+    @klass: the single dependent class if it exists
+    @x: independent columns
+    @y: dependent columns
+
+    Arguments:
+    @names: column names
+    '''
+    def __init__(self, names) -> None:
+        self.names = names 
+        self.all = []
+        self.klass = None
+        self.x = []
+        self.y = []
+
+        '''
+        First, storing all the Num and Sym Objects into the "all" attribute
+        Second, performing regex to separate the dependent from independent variables into the attributes "y" and "x"
+        respectively
+        For the atributes "x" and "y" we skip all columns which end with ":".
+        '''
+        for i in range(len(names)):
+            if(re.search("\*[A-Z]", names[i])):
```

Comment:
This doesn't appear to be matching what lua code is doing. Here, the code is matching column names which have a wildcard followed by an uppercase alphabet, while the lua code is matching the column names which are starting with an upper case alphabet. Correct me if I misunderstood this.

Reply:
On it. It is a mistake. My bad!